src/nvim/highlight.c

```diff
@@ -174,7 +174,7 @@ int ns_get_hl(NS *ns_hl, int hl_id, bool link, bool nodefault)
 {
   static int recursive = 0;
 
-  if (*ns_hl < 0) {
+  if (*ns_hl <= 0) {
```

Comment:
`ns=0` should be allowed here. It means explicitly choosing the default namespace even if some other is active. But `get_decor_provider(ns_id, true);` and below should be skipped, as `ns=0` doesn't have a provider.

Reply:
ok, I think at line 183, adding `else if (ns_id == 0) { return -1; }` should be enough.

Reply:
Ok that's actually what I tried first, so good to know I was in the right ballpark.